picard/ui/options/maintenance.py

```diff
@@ -0,0 +1,176 @@
+    NAME = "maintenance"
+    TITLE = N_("Maintenance")
+    PARENT = "advanced"
+    SORT_ORDER = 99
+    ACTIVE = True
+    HELP_URL = '/config/options_maintenance.html'
+
+    options = []
+
+    INSTRUCTIONS = N_(
+        "This allows you to remove unused option settings from the configuration INI file.\n\n"
+        "Settings that are found in the configuration file that do not appear on any option "
+        "settings page will be listed below. If your configuration file does not contain any "
+        "unused option settings, then the list will be empty and the removal checkbox will be "
+        "disabled.\n\n"
+        "Note that unused option settings could come from plugins that have been uninstalled, "
+        "so please be careful to not remove settings that you may want to use later when "
+        "the plugin is reinstalled. Options belonging to plugins that are installed but "
+        "currently disabled will not be listed for possible removal.\n\n"
+        "To remove one or more settings, first enable the removal by checking the \"Remove "
+        "selected options\" box. You can then select the settings to remove by checking the "
+        "box next to the setting. When you choose \"Make It So!\" to save your option "
+        "settings, the selected items will be removed."
+    )
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.ui = Ui_MaintenanceOptionsPage()
+        self.ui.setupUi(self)
+        self.ui.description.setText(_(self.INSTRUCTIONS))
+        self.ui.tableWidget.setHorizontalHeaderLabels([_("Option"), _("Value")])
+        self.ui.select_all.stateChanged.connect(self.select_all_changed)
+        self.ui.enable_cleanup.stateChanged.connect(self.enable_cleanup_changed)
+
+    def load(self):
+        config = get_config()
+
+        # Setting options from all option pages and loaded plugins (including plugins currently disabled).
+        key_options = set(config.setting.as_dict())
+
+        # Combine all page and plugin settings with required options not appearing in option pages.
+        current_options = set([
+            # Include options that are required but are not entered directly from the options pages.
+            'file_renaming_scripts',
+            'selected_file_naming_script_id',
+            'log_verbosity',
+            # Items missed if TagsCompatibilityWaveOptionsPage does not register.
+            'remove_wave_riff_info',
+            'wave_riff_info_encoding',
+            'write_wave_riff_info',
+        ]).union(key_options)
+
+        # All setting options included in the INI file.
+        config.beginGroup("setting")
+        file_options = set(config.childKeys())
+        config.endGroup()
+
+        orphan_options = file_options.difference(current_options)
+
+        self.ui.description.setText(
+            _(self.INSTRUCTIONS)
+            + _("\n\nThe configuration file currently contains %d option settings, %d which are unused.") % (
```

Comment:
I just noticed, but this is incorrect from i18n point of view (any simple string concatenation is).

We should rather have an extra widget for this:

```python
self.ui.config_file_contains.setText(
    _("The configuration file currently contains %d option settings, %d which are unused.") % (
        len(file_options),
        len(orphan_options)
       )
    )
) 
```

Since `self.ui.description` will remain untouched, no need to set text again for it.

Reply:
Can do, but I wonder if it would it be better to just include that line in the `INSTRUCTIONS` string.  Or do we want to keep the two separate?

Reply:
> Can do, but I wonder if it would it be better to just include that line in the `INSTRUCTIONS` string

But it wouldn't work because of https://github.com/metabrainz/picard/pull/1893/files#diff-951dd86183cd3da6256f3118e7ea870fcd39ec1e1cdeb159ff238ff8f664ea14R71

Reply:
Right.  I missed that.  I'll push the change right away.